storage/gcloud/aio/storage/bucket.py

```diff
@@ -47,10 +47,20 @@ def __init__(self, storage, name: str) -> None:
     async def list_blobs(self, prefix: str = '',
                          session: Optional[Session] = None) -> List[str]:
         params = {'prefix': prefix}
-        content = await self.storage.list_objects(self.name, params=params,
+        content = await self.storage.list_objects(self.name,
+                                                  params=params,
                                                   session=session)
-
+        next_content = {}
```

Comment:
You do a lot of juggling here with `content` and `new_content` to make sure they're formulated the right way with the right data at the right times -- as a general rule, when you find yourself mutating the same values over and over, you may be able to find a better datatype for the job. I've added a comment above which describes how you could eliminate this `next_content` variable entirely.

For interest's sake, you could also just use a simple value here; if you instead just did:
```python
next_token = content.get('nextPageToken')
while next_token:
    params['pageToken'] = next_token
    # SNIP ...
    next_token = next_content['nextPageToken']
```
then you would have a bit simpler scope -- the `next_token` value would be explicitly used as your list iterator, but `next_content` can stay as only in scope within each iteration of your list: the interesting thing there is that fewer variables needed to be juggled means your code is easier to understand; no one outside of the loop needs to care what `next_content['foobar']` is set to.

The other big benefit is that mini-refactors like this can pave the way for even bigger&better changes; in the above code snippet, now that we've replaced `next_content` with `next_token`, it might become clear that this value only ever interacts with `params['pageToken']` -- in fact, you can get away entirely without having `next_token` at all and just using `params['pageToken']` entirely! You can see how I used that trick in my above full code block to shrink the solution a bit :)